squbs-pattern/src/main/scala/org/squbs/pattern/util/ConfigUtil.scala

```
@@ -22,37 +22,46 @@ import com.typesafe.config.{Config, ConfigException, ConfigMemorySize}
-        (tag match {
-            case StringTag => Some(underlying.getString(path))
-            case StringListTag => Some(underlying.getStringList(path).toSeq)
-            case TypeTag.Int => Some(underlying.getInt(path))
-            case TypeTag.Boolean => Some(underlying.getBoolean(path))
-            case TypeTag.Double => Some(underlying.getDouble(path))
-            case ConfigTag => Some(underlying.getConfig(path))
-            case ConfigListTag => Some(underlying.getConfigList(path).toSeq)
-            case DurationTag => Some(Duration.create(underlying.getDuration(path, MILLISECONDS), MILLISECONDS))
-            case ConfigMemorySizeTag => Some(underlying.getMemorySize(path))
-            case _ => throw new IllegalArgumentException(s"Configuration option type $tag not implemented")
-          }).asInstanceOf[Option[T]]
-      else None
+    def getTry[T](path: String)(implicit tag: TypeTag[T]): Try[T] = Try {
+      (tag match {
+        case StringTag => underlying.getString(path)
+        case StringListTag => underlying.getStringList(path).toSeq
+        case TypeTag.Int => underlying.getInt(path)
+        case TypeTag.Boolean => underlying.getBoolean(path)
+        case TypeTag.Double => underlying.getDouble(path)
+        case ConfigTag => underlying.getConfig(path)
+        case ConfigListTag => underlying.getConfigList(path).toSeq
+        case FiniteDurationTag => Duration(underlying.getString(path)).asInstanceOf[FiniteDuration]
+        case DurationTag => Duration(underlying.getString(path))
+        case ConfigMemorySizeTag => underlying.getMemorySize(path)
+        case _ => throw new IllegalArgumentException(s"Configuration option type $tag not implemented")
+      }).asInstanceOf[T]
+    }
+
+    def getOption[T: TypeTag](path: String): Option[T] =
+      getTry[T](path) match {
+        case Success(value) => Some(value)
+        case Failure(e: IllegalArgumentException) => throw e
+        case Failure(e) => None
+      }
```

Thank you for feedback, I agree,  completely ignoring exception in `getOption` doesn't look like a good idea,  we should differentiate `ConfigException.Missing` from format exceptions as well. Will fix that.
The problem of injection `Logger` into `ConfigUtil` worries me a little bit. 
Making `object ConfigUtil extends LazyLogging` doesn't look perfect,  because logger's name would not be representative and helpful in localisation of exceptions. Probably, we may consider defining `ConfigUtil` as a trait with `LazyLogging` self annotation and mix it into classes instead of importing. There is a problem in this approach as well, we use either `akka.event.LoggingAdapter` or `com.typesafe.scalalogging.LazyLogging` depends on entity. What do you think about it ?